File: problem5/main.hs
myLcm_ big small inc =
    if big `mod` small == 0 then
        big
    else
        myLcm_ (big+inc) small inc

myLcm a b = myLcm_ (max a b) (min a b) (max a b)

main =
    print $ foldl myLcm 1 [1..20]
    -- print $ foldl lcm 1 [1..20]

File: problem5/main.c
#include <stdio.h>

size_t lcm(size_t a, size_t b) {
    size_t big = a > b ? b : a;
    size_t small = a > b ? a : b;

    for (size_t i = big; ; i += big) {
        if (i % small == 0)
            return i;
    }
}

int main() {

    size_t result = 1;
    for (size_t i = 2; i <= 20; i++) {
        result = lcm(i, result);
    }

    printf("Result: %d\n", result);

    return 0;
}